fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -530,6 +530,24 @@ static void clearMdc(@Nonnull Map<String, String> mdcContext) {
         }
     }
 
+    /**
+     * Get a new {@link FDBDatabaseRunner} that will run contexts similar to this one.
+     * <ul>
+     * <li>Same {@link FDBDatabase database}</li>
+     * <li>Same {@link FDBStoreTimer timer}</li>
+     * <li>Same {@link #getMdcContext() MDC context}</li>
+     * <li>Same {@link FDBDatabase.WeakReadSemantics weak read semantics}</li>
+     * </ul>
+     * @return a new database runner based on this context
+     */
+    public FDBDatabaseRunner newRunner() {
+        FDBDatabaseRunner runner = database.newRunner();
+        runner.setTimer(timer);
+        runner.setMdcContext(getMdcContext());
+        runner.setWeakReadSemantics(weakReadSemantics);
+        return runner;
```

Comment:
I suppose there probably isn't a good way of making sure that all of the things from the context get copied over to the runner (if more fields are added to a context). Oh, well.